library/graph-analytic-library/src/main/java/uk/gov/gchq/gaffer/algorithm/PageRank.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.algorithm;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import uk.gov.gchq.gaffer.operation.io.InputOutput;
+import uk.gov.gchq.gaffer.operation.serialisation.TypeReferenceImpl;
+import uk.gov.gchq.koryphe.ValidationResult;
+
+import java.util.Map;
+
+/**
+ * A {@code PageRank} is an operation which determines the PageRank values for an
+ * given input.
+ * <p>
+ * The specified input should contain a representation of {@link uk.gov.gchq.gaffer.data.element.Edge}s
+ * in a Gaffer graph. The PageRank algorithm is applied and a PageRank value is
+ * determined for each of the vertices encountered. Note that the PageRank algorithm
+ * will ignore any {@link uk.gov.gchq.gaffer.data.element.Entity} objects and values
+ * will only be calculated for entities which share a vertex with an Edge.
+ * <p>
+ * The output of a PageRank operation is identical to the input type, but the input
+ * data will be replaced with the results of the PageRank calculation.
+ * <p>
+ * Users specifying a PageRank operation MUST provide one of the following: <ul>
+ * <li>tolerance - set the desired precision of the PageRank values. With this
+ * setting PageRank will run continuously until convergence</li>
+ * <li>maxIterations - set the maximum number of iterations before
+ * returning</li> </ul>
+ * <p>
+ * Setting both of these values will result in an error.
+ */
+public class PageRank<T> implements InputOutput<T, T> {
+
+    public T input;
+    public Map<String, String> options;
+
+    /**
+     * The desired precision of the PageRank values. Cannot
+     * be used in conjunction with the maxIterations field.
+     */
+    protected Double tolerance;
+
+    /**
+     * The maximum number of iterations before terminating the operation. Cannot
+     * be used in conjunction with the tolerance field.
+     */
+    protected Integer maxIterations;
+
+    /**
+     * The probability the algorithm starts from a node chosen uniformly at random
+     * among all nodes in the network. (Also known as the damping factor.
+     */
+    protected Double resetProbability = 0.15d;
```

Comment:
Perhaps you could make 0.15d a public static constant - DEFAULT_RESET_PROBABILITY?